src/types.rs

```diff
@@ -131,6 +131,10 @@ pub struct Problem {
     pub detail: Option<String>,
     /// The HTTP status code returned for this response
     pub status: Option<u16>,
+    /// One or more subproblems associated with specific identifiers
+    ///
+    /// See <https://www.rfc-editor.org/rfc/rfc8555#section-6.7.1>
+    pub subproblems: Option<Vec<Subproblem>>,
```

Comment:
Initially I implemented this as a `Option<Vec<Problem>>` and then added an `Optional<Identifier>` field to `Problem`.

In the end I decided that wasn't a great idea because it would allow modelling sub-problems that themselves have sub-problems. Instead it felt like a dedicated `Subproblem` type that didn't admit that possibility into the type-system was preferable.

Reply:
Seems reasonable. I have a pet peeve that `Option<Vec<_>>` hardly ever seems like a reasonable type, how about expressing this as `Vec<Subproblem>` with `serde(default)`?

Reply:
> I have a pet peeve that Option<Vec<_>> hardly ever seems like a reasonable type

I was on the fence about using an empty `Vec` instead and couldn't predict which you'd like better. Switched over :+1: